core/src/banking_stage/consume_worker.rs

```diff
@@ -130,16 +133,22 @@ impl<Tx: TransactionWithMeta> ConsumeWorker<Tx> {
         Ok(())
     }
 
-    /// Try to get a bank for consuming.
-    fn get_consume_bank(&self) -> Option<Arc<Bank>> {
-        self.leader_bank_notifier
-            .get_or_wait_for_in_progress(Duration::from_millis(50))
-            .upgrade()
+    /// Get the current poh working bank with a timeout.
+    fn get_working_bank_with_timeout(&self) -> Option<Arc<Bank>> {
```

Comment:
nit: we must kill the get_ prefixes at some point (thing(), set_thing() is
idiomatic rust)